bolt-cli/src/common/web3signer.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    collections::HashMap,
+    fs::{self, File},
+    io::Read,
+};
+
+use crate::cli::TlsCredentials;
+use eyre::{Context, ContextCompat, Result};
+use reqwest::{Certificate, Identity, Url};
+use serde::Deserialize;
+
+/// Web3Signer remote server.
+///
+///  Available services:
+/// - `Lister`: List accounts in the keystore.
+/// - `Signer`: Request a signature from the remote signer.
+/// - `AccountManager`: TODO
+///
+/// Reference: https://docs.web3signer.consensys.io/reference
+#[derive(Clone)]
+pub struct Web3Signer {
+    base_url: Url,
+    client: reqwest::Client,
+}
+
+/// Keys represents the indiviudal object returned by commitboost.
+///
+/// Reference: https://commit-boost.github.io/commit-boost-client/api/
+#[derive(Deserialize, Debug)]
+struct Keys {
+    pub consensus: String,
+    proxy_bls: Vec<String>,
+    proxy_ecdsa: Vec<String>,
+}
+
+/// CommitBoostKeys is the container type for `get_pubkeys` endpoint.
+///
+/// /// Reference: https://commit-boost.github.io/commit-boost-client/api/
+#[derive(Deserialize, Debug)]
+struct CommitBoostKeys {
+    keys: Vec<Keys>,
+}
+
+impl Web3Signer {
+    pub async fn connect(addr: String, credentials: TlsCredentials) -> Result<Self> {
+        // Establish connection with TLS config.
+        let base_url = addr.parse()?;
+        let (cert, identity) = compose_credentials(credentials)?;
+        let client = reqwest::Client::builder()
+            // TODO: handle this properly
```

Comment:
Need to figure out why TLS is not working with `reqwest`

Reply:
Fixed https://github.com/chainbound/bolt/pull/513/commits/0461b6903ff7531ae3522e30fa3dfe6221ed02e3